src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -988,9 +993,10 @@ def do_resolve_with_resolved_children(
                 if logical_plan.format_type_options
                 else {}
             )
-            format_name = logical_plan.cur_options.get("FORMAT_NAME")
+            format_name = (logical_plan.cur_options or {}).get("FORMAT_NAME")
```

Comment:
Looking at the only reference to `CopyIntoTable` at https://github.com/snowflakedb/snowpark-python/blob/697294d8ee89396deab4309dea384b1eadb75d6b/src/snowflake/snowpark/dataframe.py#L2819, `cur_options` will never be None.

Reply:
Please can you take over the PR where possible?

Reply:
Will do, thank you for making this happen, really appreciate your work here! 🎉